modules/caption-impl/src/main/java/org/opencastproject/caption/converters/GoogleSpeechCaptionConverter.java

```diff
@@ -0,0 +1,188 @@
+                indexLast = j;
+                // Create a caption
+                double start = -1;
+                double end = -1;
+                if (indexLast < timestampsArray.size()) {
+                  // Get start time of first element
+                  JSONObject wordTSList = (JSONObject) timestampsArray.get(indexFirst);
+                  if (wordTSList.size() == 3) {
+                    // Remove 's' at the end
+                    Number startNumber = NumberFormat.getInstance().parse(removeCharacter((wordTSList.get("startTime").toString()), 's'));
+                    start = startNumber.doubleValue();
+                  }
+                  // Get end time of last element
+                  wordTSList = (JSONObject) timestampsArray.get(indexLast);
+                  if (wordTSList.size() == 3) {
+                    Number endNumber = NumberFormat.getInstance().parse(removeCharacter((wordTSList.get("endTime").toString()), 's'));
+                    end = endNumber.doubleValue();
+                  }
+                }
+                if (start == -1 || end == -1) {
+                  logger.warn("Could not build caption object for job {}, result index {}: start/end times not found",
+                          jobId, i);
+                  continue resultsLoop;
+                }
+
+                String[] captionLines = new String[1];
+                captionLines[0] = line.toString().replace("%HESITATION", "...");
+                captionList.add(new CaptionImpl(buildTime((long) (start * 1000)), buildTime((long) (end * 1000)),
+                        captionLines));
+                indexFirst = -1;
+                indexLast = -1;
+                line.setLength(0);
+              }
+            }
+          }
+        }
+      }
+    } catch (Exception e) {
+      logger.warn("Error when parsing Google transcriptions result: {}" + e.getMessage());
+      throw new CaptionConverterException(e);
+    }
+
+    return captionList;
+  }
+
+  @Override
+  public void exportCaption(OutputStream outputStream, List<Caption> captions, String language) throws IOException {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public String[] getLanguageList(InputStream inputStream) throws CaptionConverterException {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public String getExtension() {
+    return "json";
+  }
+
+  @Override
+  public Type getElementType() {
+    return MediaPackageElement.Type.Attachment;
+  }
+
+  private Time buildTime(long ms) throws IllegalTimeFormatException {
+    int h = (int) (ms / 3600000L);
+    int m = (int) ((ms % 3600000L) / 60000L);
+    int s = (int) ((ms % 60000L) / 1000L);
+    ms = (int) (ms % 1000);
+
+    return new TimeImpl(h, m, s, (int) ms);
+  }
+
+  private String removeCharacter(String str, char car) {
+    if (str != null && str.length() > 0 && str.charAt(str.length() - 1) == car) {
```

Comment:
"car" for character :-) , this code is to remove last character from the provided google timestamp which has the format: 100s.  See line 117 on how it's used: (removeCharacter((wordTSList.get("endTime").toString()), 's'))
@gregorydlogan  do you want me to write it differently?

Reply:
@tyfranck My initial comment was my suggestion :)

You're checking if it's null, that its length is greater than zero, and then if the last character matches.  I'm checking if it's null, then if it ends with a given character.  I would even simplify my initial comment with

`if (str.endsWith(car))`, since I'm 99% sure `.endsWith` checks for nulls.

Reply:
I get it now :-) . Completely misread it.

Reply:
fixed